Lib/cgi.py

```diff
@@ -661,11 +661,9 @@ def read_multi(self, environ, keep_blank_values, strict_parsing):
 
     def read_single(self):
         """Internal: read an atomic part."""
+        self.read_lines()
```

Comment:
I just tested this patch locally. Seems like with some clients, the webserver will not get a EOF if the client sent content-length header. So the this will result in an indefinite block on `fp.readline()`. We should honor the content-length, and not try to read more than that in any case.